packages/financial-templates-lib/src/price-feed/DefiPulseTotalPriceFeed.js

```diff
@@ -0,0 +1,151 @@
+const { PriceFeedInterface } = require("./PriceFeedInterface");
+const assert = require("assert");
+
+// An implementation of PriceFeedInterface that uses DefiPulse Data api to retrieve prices.
+class DefiPulseTotalPriceFeed extends PriceFeedInterface {
+  /**
+   * @notice Constructs the DefiPulseTVL_ALLPriceFeed.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} web3 Provider from truffle instance to connect to Ethereum network.
+   * @param {String} apiKey DeFiPulse Data API key. Note: these API keys are rate-limited.
+   * @param {Integer} lookback How far in the past the historical prices will be available using getHistoricalPrice.
+   * @param {Object} networker Used to send the API requests.
+   * @param {Function} getTime Returns the current time.
+   * @param {Integer} minTimeBetweenUpdates Min number of seconds between updates. If update() is called again before
+   *      this number of seconds has passed, it will be a no-op.
+   */
+  constructor(logger, web3, apiKey, lookback, networker, getTime, minTimeBetweenUpdates, priceFeedDecimals) {
```

Comment:
Please see @chrismaree's comments around providing a priceFeedDecimals param that defaults to 18 and allows the user to output values with different numbers of decimals based on that param. See the cryptowatch price feed for an example of how this can work.